compiler/cli/cli-base/src/org/jetbrains/kotlin/cli/jvm/modules/CliJavaModuleFinder.kt

```diff
@@ -119,7 +120,13 @@ class CliJavaModuleFinder(
     private fun findSystemModule(moduleRoot: VirtualFile, useSig: Boolean = false): JavaModule.Explicit? {
         val file = moduleRoot.findChild(if (useSig) PsiJavaModule.MODULE_INFO_CLASS + ".sig" else PsiJavaModule.MODULE_INFO_CLS_FILE)
             ?: return null
-        val moduleInfo = JavaModuleInfo.read(file, javaFileManager, allScope) ?: return null
+        val moduleInfo = try {
+            JavaModuleInfo.read(file, javaFileManager, allScope) ?: return null
+        } catch (e: IllegalStateException) {
```

Comment:
It's usually not a good idea to catch such a broad exception type. Please create a new exception near the place where it's thrown (e.g. a nested class `JavaModuleInfo.ReadingException`), inherit it from `IllegalStateException`, and catch it here instead.